inc/saiswitch.h

```diff
@@ -3048,6 +3048,33 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_MAX_ICMP_ECHO_SESSION,
 
+    /**
+     * @brief Maximum size of the packet after trimming
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 128
+     */
+    SAI_SWITCH_ATTR_PACKET_TRIMMING_SIZE,
+
+    /**
+     * @brief New packet trimming DSCP value
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_SWITCH_ATTR_PACKET_TRIMMING_DSCP_VALUE,
+
+    /**
+     * @brief New packet trimming queue index
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_SWITCH_ATTR_PACKET_TRIMMING_QUEUE_INDEX,
```

Comment:
@marian-pritsak How does the NOS knows what queue index to program in this case? Capability query will not return a support queue index value. We should let HW decide what queue index it will support for TRIM and make this attribute as READ_ONLY.

Else please provide a workflow that how NOS is supposed to know the queue index

Reply:
Added two modes - mapping from QoS maps and straight setting with this attribute.